cmd/kubelet/app/plugins.go

```diff
@@ -67,26 +62,21 @@ func ProbeVolumePlugins() []volume.VolumePlugin {
 	//
 	// Kubelet does not currently need to configure volume plugins.
 	// If/when it does, see kube-controller-manager/app/plugins.go for example of using volume.VolumeConfig
-	allPlugins = append(allPlugins, awsebs.ProbeVolumePlugins()...)
+	allPlugins = appendLegacyProviderVolumes(allPlugins)
```

Comment:
`appendLegacyProviderVolumes` for cmd/kubelet and cmd/kube-controller-manager are quite similar. In https://github.com/kubernetes/kubernetes/pull/80587/files, I tried to merge them in a new pkg/volume/util/csimigration/plugins.go file that both kubelet and kcm calls.

Once that PR lands, I think we can remove cmd/kubelet/app/plugins_legacyproviders.go and cmd/kube-controller-manager/app/plugins_legacyproviders.go (along with their _nolegacyproviders.go files) here in favor of a pkg/volume/util/csimigration/legacy_plugins.go / no_legacy_plugins.go.

Reply:
SGTM, I tried to keep refactors minimal and local to each package for the purposes of this PR, but I spoke to @davidz627 about something similar to this as a follow up (though apparently it might land first! :upside_down_face:)

Reply:
@BenTheElder https://github.com/kubernetes/kubernetes/pull/80587 is the one we discussed that @ddebroy is actually working on :)